attachment_s3/__init__.py

```diff
@@ -1,2 +1,18 @@
-
+# Copyright 2016-2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+import os
 from . import models
+from . import controllers
+
+from odoo import api, SUPERUSER_ID
+
+
+def _post_init_hook(cr, registry):
+    # create the S3 bucket after module installation
+    if os.environ.get("AWS_BUCKETNAME", False):
```

Comment:
regarding this post init and also de "restore" we're working in a solution that may be could help here.
The problem we face is that if you call "force_storage" for big databases, it will take a lot of time.
So we're doing a simple modification and adding a cron (Disable by default) that you can enable and then it starts to move the filestore in batch to s3.
With that optional functionality, this bot use cases would be solved.

Have you try to upload big databases? It works ok with normal odoo timeout limits?

Reply:
@jjscarafia We haven't tested that scenario yet.